### ISSUE TITLE
Rewrite outlook safelink URLs

### ISSUE THREAD

**Issue by austinhuang0131** [apparition47/MailTrackerBlocker#112]
I use an Office 365 inbox and all my links are rewritten like this:

https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Faustinhuang0131%2F&data=04%7C01%7Caustinhuang%40sampleemailaddress.com%7Cb5f7ed0cd3ca4977a2d808d914a3b367%7C81824d4b6a574db49a15217b19b2f9e9%7C0%7C0%7C637563515598406011%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=y2j9T1cvMgtQNR29%2Bn8Svj0PNorZSZS7J2ZQJoNwV84%3D&reserved=0

When `sampleemailaddress` is replaced with my actual email address, it will visit the URL as indicated in the `url` parameter. So is it possible to extract that parameter and undo the rewrite? Thanks.

**Comment by apparition47** [comment]
This is a feature request and not a bug with MailTrackerBlocker right?

I've been looking into detracking <a> links. Looks like it's possible to do here but not in all cases because the redirect link is sometimes encoded then scrambled. May need another set of rules for this one.

Will keep this in mind, thanks for sharing that URL.

**Comment by austinhuang0131** [comment]
Yeah that's an enhancement, sorry for not indicating

This one is pretty clear, you can also look up on GitHub for more examples of this

**Comment by apparition47** [comment]
Yep, yours is clear but I've seen clickable hyperlinks that look like this where it isn't possible (e.g. SendGrid):

```
http://url9216.customers.gumroad.com/ls/click?upn=lxUZP-2FJG4eH36wcqVLdRrts4ZQQmWiM-2FBszgXWRtSoIlCchLU8TftB45jSHKRZZCbZrlIEHwZr-2FNsvxBm-2F2bYw-3D-3DUBQg_DtJU5KezLEf9Vl9dVV-2BNPAOv1VBd-2BHw4r3Vy6FLArsnkp2RxK5yBFTSJ3BPedpa9pppz2BX23sqnzmjt48Cdu-2B42fUVU1e9d5AflLTW2fgw4X39HJFTlZg8Mlcekh1nlIwt3-2F1mO7n4R0EJEj7AJdtvjg34UZZVTFFXqyN4a1wQJeR0vSExNEGdHMI-2BNg1wdcNgwLuphthFdob-2FDcvfjhp7ESHJ-2F5-2FfwM5TGcoWVRYxO99ks1VAqMSKbxbf3hpJbOQi0O5W37jWEXIcbKmCFLEXd2iKssTDo3X5KGY09dT-2FDpxWGh7XXNsRFO1uktDmADaAW-2FYII0QJC3Ev8ct5Ltb7KQ5EhMrLJGI9yay19I4eQ3pXhpyi7yL-2BBpzdDCrioBusnohircnPwC9w4yw-3D-3D
```

**Comment by m-schmitt** [comment]
safelinks.protection.outlook.com isn't for tracking, it is malware protection. And I think you can disable it in your Outlook.com settings.

**Comment by austinhuang0131** [comment]
@m-schmitt i don't think you can disable this on Office 365 (exchange) inboxes by yourself (not by company admin) (but i don't use it anymore so idrc at this point so correct me if i'm wrong)

**Comment by austinhuang0131** [comment]
Reddit:

```
https://click.redditmail.com/CL0/https:%2F%2Fwww.reddit.com%2F/1/0100017c1a15a5b9-b8bb47fa-fa9d-4e80-aa6d-2f61353ce624-000000/Ye5qiCp0_4SpV1DQXcDKNMvFsvpmtWmqvvqYacdIeK8=216
```

Middle path, so `https://www.reddit.com/`
